doc/source/dev/core-dev/deprecations.rst.inc

```diff
@@ -7,25 +7,41 @@ There are various reasons for wanting to remove existing functionality: it's
 buggy, the API isn't understandable, it's superseded by functionality with
 better performance, it needs to be moved to another SciPy submodule, etc.
 
-In general it's not a good idea to remove something without warning users about
-that removal first.  Therefore this is what should be done before removing
+In general, it's not a good idea to remove something or make other backwards
+compatibility breaking changes without warning users about
+the change first. Therefore this is what should be done before deprecating
 something from the public API:
 
 #. Propose to deprecate the functionality on the scipy-dev mailing list and get
    agreement that that's OK.
```

Comment:
An excerpt of the discussion from https://github.com/scipy/scipy/issues/14714 that leads me to the following suggestion:

> @h-vetinari: I think it would be good idea to have something of a standing policy that if we're touching the signature of a function in a way that needs a deprecation cycle (e.g. remove/rename a keyword) to also use that to turn the keywords into keyword-only arguments, but for now I'll just try to remember that when I come across a case.

> @tupui: @h-vetinari feel free to put this on the agenda of our next community meeting. [note: AFAIK this never happened]

> @rgommers: Adding a note to http://scipy.github.io/devdocs/dev/api-dev/api-dev-toc.html about positional-only and keyword-only arguments sounds like a good idea to me.
> 
> Re wholesale changes: that's going to break a _ton_ of code, which isn't justified.

I think the docs have shifted a bunch since then, so this is a good place IMO. We could still add something separately that _new_ functions should prefer keyword-only arguments as well.


```suggestion
#. Propose to deprecate the functionality on the scipy-dev mailing list and get
   agreement that that's OK.
#. When changing a function signature (e.g. deprecating a keyword argument), this
   opportunity should be used to convert that function to keyword-only usage,
   see GH-14714.
```

Reply:
> I think the docs have shifted a bunch since then, so this is a good place IMO. We could still add something separately that _new_ functions should prefer keyword-only arguments as well.

I think this covers this: https://docs.scipy.org/doc/scipy/dev/missing-bits.html#required-keyword-names